modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -63,47 +62,45 @@
     private static Log log = LogFactory.getLog(AndesKernelBoot.class);
     private static VirtualHost virtualHost;
     private static MessageStore messageStore;
-
     /**
      * Scheduled thread pool executor to run periodic andes recovery task
      */
     private static ScheduledExecutorService andesRecoveryTaskScheduler;
-
+    /**
+     * Scheduled thread pool executor to run periodic expiry message deletion task
+     */
+    private static ScheduledExecutorService expiryMessageDeletionTaskScheduler;
     /**
      * Used to get information from context store
      */
     private static AndesContextStore contextStore;
-
     /**
      * This is used by independent worker threads to identify if the kernel is performing shutdown operations.
      */
     private static boolean isKernelShuttingDown = false;
-
     /**
      * This will boot up all the components in Andes kernel and bring the server to working state
      */
     public static void initializeComponents() throws AndesException {
         isKernelShuttingDown = false;
-
         //loadConfigurations - done from outside
         //startAndesStores - done from outside
         int threadPoolCount = 1;
         ThreadFactory namedThreadFactory = new ThreadFactoryBuilder()
                 .setNameFormat("AndesRecoveryTask-%d").build();
         andesRecoveryTaskScheduler = Executors.newScheduledThreadPool(threadPoolCount);
+        expiryMessageDeletionTaskScheduler = Executors.newScheduledThreadPool(threadPoolCount);
```

Comment:
Would it make sense to have a different named thread factory for the `expiryMessageDeletionTaskScheduler`. Having the thread named as `AndesRecoveryTask` is wrong IMO.